lib/notifiers/delivery_notifier.dart

```diff
@@ -18,12 +18,17 @@ class DeliveryNotifier extends ChangeNotifier {
       return;
     }
 
-    final delivery = await _repository.getCurrentDelivery(
-      entityId: user.entityId,
-      time: DateTimeUtils.getDateTimeOfCurrentDelivery(user.pickUpFrom),
-    );
-    _delivery = delivery;
-    notifyListeners();
+    _repository
+        .observeDelivery(entityId: user.entityId)
+        .listen((deliveries) async {
+      final delivery = await _repository.getCurrentDelivery(
```

Comment:
Iam thinking about this. I would probably prefer to see the observation properly seeded to a specific delivery at this point, so it doesn't have to be queried again. Do you think it would be ok to use the last delivery for the entity?

Reply:
Hmm, I am not sure if this would work correctly on production, where deliveries are created 1 day ahead. I think user will always end up observing tomorrow's delivery 🤔 

On DEV it works OK, as the CRON is not running and new empty deliveries are not created.

Anyway I guess you could simply replace the `DeliveryService::getDelivery` function with an observable one:
```dart
Stream<DeliveryDto?> observeDelivery(String donorId, DateTime time) {
  final snapshots = _collection
      .where('donorId', isEqualTo: donorId)
      .where('type', isEqualTo: DeliveryTypeDto.foodDelivery.toJson())
      .whereTime('pickupTimeWindow.start', time)
      .snapshots();
  return snapshots.map((snapshot) => snapshot.docs.firstOrNull?.data());
}
``` 

And use it in this notifier
```dart
_repository.observeCurrentDelivery(
  entityId: user.entityId,
  time: DateTimeUtils.getDateTimeOfCurrentDelivery(user.pickUpFrom),
).listen((delivery) {
  _delivery = delivery;
  notifyListeners();
});
```

That should work even when there is some newer delivery is created for tomorrow. 

Also I am not sure if we should cancel the subscription when the new one is created 🤔 
Something like this for example:

```dart
StreamSubscription<Delivery?>? _streamSubscription;

void init(UserData user) async {
   // ...
  _streamSubscription?.cancel();
  _streamSubscription = _repository.observeCurrentDelivery(...)
}
```

Reply:
Yes, that looks neat, Ihave modified it. Thanks